src/commands/layout/push.ts

```diff
@@ -0,0 +1,110 @@
+import { Args, Flags } from "@oclif/core";
+
+import BaseCommand from "@/lib/base-command";
+import { KnockEnv } from "@/lib/helpers/const";
+import { formatError, formatErrors, SourceError } from "@/lib/helpers/error";
+import * as CustomFlags from "@/lib/helpers/flag";
+import { merge } from "@/lib/helpers/object";
+import { formatErrorRespMessage, isSuccessResp } from "@/lib/helpers/request";
+import { indentString } from "@/lib/helpers/string";
+import { spinner } from "@/lib/helpers/ux";
+import * as EmailLayout from "@/lib/marshal/email-layout";
+import { WithAnnotation } from "@/lib/marshal/shared/types";
+
+export default class EmailLayoutPush extends BaseCommand<
+  typeof EmailLayoutPush
+> {
+  static aliases = ["email-layout:push", "email_layout:push"];
+
+  static summary =
+    "Push one or more email layouts from a local file system to Knock.";
+
+  static flags = {
+    environment: Flags.string({
+      summary:
+        "Pushing an email layout is only allowed in the development environment",
+      default: KnockEnv.Development,
+      options: [KnockEnv.Development],
+    }),
+    all: Flags.boolean({
+      summary: "Whether to push all layouts from the target directory.",
+    }),
+    "layouts-dir": CustomFlags.dirPath({
+      summary: "The target directory path to find all layouts to push.",
+      dependsOn: ["all"],
+    }),
+    commit: Flags.boolean({
+      summary: "Push and commit the layout(s) at the same time",
+    }),
+    "commit-message": Flags.string({
+      summary: "Use the given value as the commit message",
+      char: "m",
+      dependsOn: ["commit"],
+    }),
+  };
+
+  static args = {
+    emailLayoutKey: Args.string({
+      required: false,
+    }),
+  };
+
+  async run(): Promise<void> {
+    const { flags } = this.props;
+
+    // 1. First read all layout directories found for the given command.
+    const target = await EmailLayout.ensureValidCommandTarget(
+      this.props,
+      this.runContext,
+    );
+
+    const [layouts, readErrors] = await EmailLayout.readAllForCommandTarget(
+      target,
+      { withExtractedFiles: true },
+    );
+
+    if (readErrors.length > 0) {
+      this.error(formatErrors(readErrors, { prependBy: "\n\n" }));
+    }
+
+    if (layouts.length === 0) {
+      this.error(`No layout directories found in ${target.context.abspath}`);
+    }
+
+    // 2.TODO: Validate the layouts before upserting
```

Comment:
I'll add the validation on the following PR, which will also add the validation endpoint.